### ISSUE TITLE
How to properly set up XLFormRowDescriptorTypeSelectorPush?

### ISSUE THREAD

**Issue by conradwt** [xmartlabs/XLForm#537]
Hi, I have gone through both the code and the documentation in order to setup XLFormRowDescriptorTypeSelectorPush without any success.  At this time, I have the following:

Here's the initial form:

```swift
class NewSnapshotViewController : XLFormViewController {
  
  var context = CoreDataManager.managedObjectContext!
  
  required init(coder aDecoder: NSCoder) {
    super.init(coder: aDecoder)
    self.initializeForm()
  }
  
  override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
    super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    self.initializeForm()
  }

  override func viewDidLoad() {
    super.viewDidLoad()
  }
  
  override func viewWillAppear(animated: Bool) {
    super.viewWillAppear(animated)
  }
  
  func initializeForm() {
    var form : XLFormDescriptor
    var section : XLFormSectionDescriptor
    var row : XLFormRowDescriptor
    
    form = XLFormDescriptor(title: "New Entry") as XLFormDescriptor
    
    // Name Section
    section = XLFormSectionDescriptor.formSectionWithTitle("") as XLFormSectionDescriptor
    form.addFormSection(section)
    
    row = XLFormRowDescriptor(tag: "selectedChild", rowType: XLFormRowDescriptorTypeSelectorPush, title: "One")
    row.action.viewControllerStoryboardId = "SelectOneViewController"
    section.addFormRow(row)
    
    row = XLFormRowDescriptor(tag: "note", rowType: XLFormRowDescriptorTypeTextView, title: "Two")
    section.addFormRow(row)
    
    row = XLFormRowDescriptor(tag: "selectedMilestone",  rowType: XLFormRowDescriptorTypeSelectorPush, title: "Three")
    row.action.viewControllerStoryboardId = "SelectTwoViewController"
    section.addFormRow(row)
    
    self.form = form
  }

}

```

Here are the controllers where a user can select a single item to populate the above controller.

```swift
class SelectOneViewController : UITableViewController {
  ...
}

class SelectTwoViewController : UITableViewController {
  ...
}
```

**Comment by mats-claassen** [comment]
Hi, your ViewControllers have to conform to `XLFormRowDescriptorViewController` protocol.
Did you make sure that the StoryboardId's are correct?

**Comment by conradwt** [comment]
@mats-claassen Both SelectOneViewController and SelectTwoViewController are using UITableViewController components on the storyboard.  What does it mean to conform to this protocol beyond adding the one required property?  It may be a good idea to create a cookbook style for the FAQ where we group related questions as chapters to provide better structure for the README.  For example,

### Q. 1.  How do use an XLFormRowDescriptorTypeSelectorPush with an existing dynamic  UITableViewController source class?

### A. 1.  ...

**Comment by frapaa** [comment]
Use `class SelectOneViewController : UITableViewController <XLFormRowDescriptorViewController>` 

then `@synthesize rowDescriptor;` 
and in `- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath` set the value: `rowDescriptor.value = [entries objectAtIndex:indexPath.row];`

then just pop back to the form.

**Comment by conradwt** [comment]
@fcanas Yes, I found a complete Objective-C example about a day or so go which provided the provided the solution for setting all this up.  BTW, I followed the Simpsons example from the Swift Examples Dynamic Selectors from the following location:

```
Examples/Swift/Examples/Selectors/DynamicSelector
```

However, both Swift files were written Objective-C.  Thus, my Objective-C past is starting to pay off when moving logic from Objective-C to Swift.
